From frankpzh@gmail.com  Tue Jun  1 13:44:47 2010
From: Frank Pan <frankpzh@gmail.com>
Date: Wed, 26 May 2010 15:37:43 +0800
Subject: tty: fix a little bug in scrup, vt.c
To: Andrew Morton <akpm@linux-foundation.org>, Greg Kroah-Hartman <gregkh@suse.de>, Catalin Marinas <catalin.marinas@arm.com>, Alan Cox <alan@linux.intel.com>, Daniel Mack <daniel@caiaq.de>
Cc: linux-kernel@vger.kernel.org
Message-ID: <AANLkTil5HfBzgN-3ttSOUfQXxssR06nyIl1UoC9-zCiA@mail.gmail.com>


The code uses vc->vc_cols instead of vc->vc_size_row by mistake, it
will cause half of the region which is going to clear remain
uncleared.
The issue happens in background consoles, so it's hard to observe.

Frank Pan

Signed-off-by: Frank Pan <frankpzh@gmail.com>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/vt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/char/vt.c
+++ b/drivers/char/vt.c
@@ -304,7 +304,7 @@ static void scrup(struct vc_data *vc, un
 	d = (unsigned short *)(vc->vc_origin + vc->vc_size_row * t);
 	s = (unsigned short *)(vc->vc_origin + vc->vc_size_row * (t + nr));
 	scr_memmovew(d, s, (b - t - nr) * vc->vc_size_row);
-	scr_memsetw(d + (b - t - nr) * vc->vc_cols, vc->vc_video_erase_char,
+	scr_memsetw(d + (b - t - nr) * vc->vc_size_row, vc->vc_video_erase_char,
 		    vc->vc_size_row * nr);
 }
 
